src/plugins/dashboard/public/application/legacy_app.js

```diff
@@ -171,7 +170,12 @@ export function initDashboardApp(app, deps) {
             history.push(deps.addBasePath(viewUrl));
           };
           $scope.delete = (dashboards) => {
-            return service.delete(dashboards.map((d) => d.id));
+            const ids = dashboards.map((d) => ({ id: d.id, appId: d.appId }));
+            return Promise.all(
+              ids.map(({ id, appId }) => {
+                return deps.savedObjectsClient.delete(appId, id);
+              })
+            );
```

Comment:
For consistency with the visualizations list, we should at least fire a toast if there's a deletion error:
```suggestion
            return Promise.all(
              ids.map(({ id, appId }) => {
                return deps.savedObjectsClient.delete(appId, id);
              })
            ).catch((error) => {
              toastNotifications.addError(error, {
                title: i18n.translate('dashboard.dashboardListingDeleteErrorTitle', {
                  defaultMessage: 'Error deleting dashboard',
                }),
              });
            });
```